### PR TITLE
remove jQuery

### PR DESCRIPTION
**Pull request by shaundillon**
Does as it says on the tin. Removes the need for jQuery. The other changes are due to restrictions in JQLite that comes with angular. Here's [the API ref](https://docs.angularjs.org/api/ng/function/angular.element#angularjs-s-jqlite) to clarify

### CODE REVIEW

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js [hunk 1]
```diff
@@ -192,7 +192,11 @@ grCollectionsPanel.directive('grNode', ['$parse', '$compile', function($parse, $
                 }
 
                 compiledTemplate(scope, cloned => {
-                    element.find('.node__children').append(cloned);
+                    const container = element.find('gu-template-container');
+                    // JQLite doesn't have modern array methods :(
+                    for (let i = 0; i < cloned.length; i++) {
```
Comment (akash1810) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: could do `Array.from(cloned).forEach(foo => bar)`?

File: kahuna/public/js/components/gu-date-range/gu-date-range.js [hunk 2]
```diff
@@ -53,126 +53,125 @@ guDateRange.directive('guDateRange', [function () {
...
-            $scope.save = setCustomRange;
-            $scope.setPresetDate = setPresetDate;
-            $scope.clearStart = clearStart;
-            $scope.clearEnd = clearEnd;
-
-            resetView();
+          ctrl.guDateFormat = ctrl.guDateFormat || 'DD-MMM-YYYY';
+          ctrl.guAnyTimeText = ctrl.guAnyTimeText || 'anytime';
+          ctrl.guFirstDay = ctrl.guFirstDay || 0;
+          var startInput = el[0].querySelectorAll('.gu-date-range__input__start--hidden')[0];
```
Comment (akash1810) [inline kahuna/public/js/components/gu-date-range/gu-date-range.js]: minor: worth creating a variable for `el[0]` for DRYness?

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js (hunk 1 above)
Comment (shaundillon) [reply kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: Good point!

File: kahuna/public/js/components/gu-date-range/gu-date-range.js (hunk 2 above)
Comment (shaundillon) [reply kahuna/public/js/components/gu-date-range/gu-date-range.js]: Yep, sounds good to me. It's a pattern used in the main.js file too.

File: kahuna/public/js/components/gu-date-range/gu-date-range.js [hunk 3]
```diff
@@ -53,126 +53,130 @@ guDateRange.directive('guDateRange', [function () {
...
-            $scope.save = setCustomRange;
-            $scope.setPresetDate = setPresetDate;
-            $scope.clearStart = clearStart;
-            $scope.clearEnd = clearEnd;
-
-            resetView();
+          ctrl.guDateFormat = ctrl.guDateFormat || 'DD-MMM-YYYY';
+          ctrl.guAnyTimeText = ctrl.guAnyTimeText || 'anytime';
+          ctrl.guFirstDay = ctrl.guFirstDay || 0;
+          const originalEl = el[0];
```
Comment (shaundillon) [inline kahuna/public/js/components/gu-date-range/gu-date-range.js]: This is where the actual change starts, the rest is autoindent

**Pull request merged**